rasa/shared/core/flows/flows_yaml_schema.json

```diff
@@ -0,0 +1,302 @@
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "next"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "next": {
+                "$ref": "#$defs/next"
+              },
+              "id": {
+                "type": "string"
+              }
+            }
+          },
+          {
+            "required": [
+              "generation_prompt"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "generation_prompt": {
+                "type": "string"
+              },
+              "id": {
+                "type": "string"
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "intent"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "intent": {
+                "type": "string"
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          },
+          {
+            "required": [
+              "entry_prompt"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "id": {
+                "type": "string"
+              },
+              "entry_prompt": {
+                "type": "string"
+              },
+              "next": {
+                "$ref": "#$defs/next"
+              }
+            }
+          }
+        ]
+      }
+    },
+    "flow": {
+      "required": [
+        "steps"
+      ],
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "if": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "nlu_trigger": {
+          "type": "array",
+          "items": {
+            "required": [
+              "intent"
+            ],
+            "type": "object",
+            "properties": {
+              "intent": {
+                "type": "object",
+                "properties": {
+                  "confidence_threshold": {
+                    "type": "number"
+                  },
+                  "name": {
+                    "type": "string"
+                  }
+                },
+                "required": [
+                  "name"
+                ]
+              }
+            }
+          }
+        },
+        "steps": {
+          "$ref": "#$defs/steps"
+        }
+      }
+    },
+    "next": {
+      "anyOf": [
+        {
+          "type": "array",
+          "minContains": 1,
+          "items": {
+            "type": "object",
+            "oneOf": [
```

Comment:
More to clarify my own understanding of the new nested flow format, if next is specified, it can only be specified with if-then or else, otherwise we can omit it, in which case it will point to the next in the sequence?

Reply:
Im still figuring out the format myself 🥲 from what I understood from the flow spec `next` can be either a `branch` with the `if/else` or a `string` with values of `END` or a step `ID`  not sure what you mean with the omit part.

Reply:
> not sure what you mean with the omit part.

One of the benefits of the new format was to allow `next` and `id` step keys to be optional. When they're missing in a flow step, they get resolved internally. I think your explanation above answers my question.